packages/meteor/asl-helpers.js

```diff
@@ -6,16 +6,12 @@ Meteor._isFibersEnabled = !process.env.DISABLE_FIBERS && Meteor.isServer;
 Meteor._getAslStore = getAslStore;
 Meteor._getValueFromAslStore = getValueFromAslStore;
 Meteor._updateAslStore = updateAslStore;
+if (Meteor.isServer && !global.asyncLocalStorage) {
+    const { AsyncLocalStorage } = Npm.require('async_hooks');
```

Comment:
> check if fibers are enabled

If you are creating something with Meteor 3, it'll be implicit that Fibers is always disabled, right? So, I think, you'll always want to have this.

Reply:
Build plugins can run in older versions of Meteor, but frequently include dependencies from newer versions of Meteor. To support Meteor 3, they will have to use versions of dependencies from Meteor 3 to ensure they don't use fibers, and use those same versions in Meteor 2 which has fibers enabled.

For example, I want `zodern:melte` to work with Meteor 1.8.1 - Meteor 3. Since it uses async/await, it will need to be published with Meteor 3 to ensure it uses native async/await, and will include the versions of meteor, babel-compiler, and ecmascript from meteor 3. When the published package is run in Meteor 1 or 2, fibers will be enabled.

I did some of the work for this in https://github.com/meteor/meteor/pull/12537. The async hooks is the other main issue I am aware of that would need to be addressed.